ui/client/common/ProcessStateUtils.js

```diff
@@ -1,39 +1,116 @@
+    NOT_DEPLOYED: "Process has been never deployed.",
+    DURING_DEPLOY: "Process has been already started and currently is during deploying.",
+    RUNNING: "Process successful deployed and currently is running.",
+    CANCELED: "Process is successfully canceled.",
+    RESTARTING: "Process is restarting...",
+    FAILED: "There are some problems with process..",
+    FINISHED: "Process completed successfully.",
+  },
+
+  STATUS_ICONS: {
+    UNKNOWN: InlinedSvgs.iconUnknown,
+    NOT_DEPLOYED: InlinedSvgs.iconNotDeployed,
+    DURING_DEPLOY: InlinedSvgs.iconDeployRunningAnimated,
+    RUNNING: InlinedSvgs.iconRunningAnimated,
+    CANCELED: InlinedSvgs.iconStoppingSuccess,
+    RESTARTING: InlinedSvgs.iconStoppedWorking,
+    FAILED: InlinedSvgs.iconFailed,
+    FINISHED: InlinedSvgs.iconSuccess,
+  },
+
+  isRunning(state) {
+    return state != null && state.status.toUpperCase() === this.STATUSES.RUNNING
+  },
+
+  isStopped(state) {
+    return state != null && state.status.toUpperCase() === this.STATUSES.CANCELED
+  },
 
   isDeployed(process) {
     const action = _.get(process, 'deployment.action')
-    return action != null && action.toUpperCase() === this.ACTION_DEPLOY
+    return action != null && action.toUpperCase() === this.ACTIONS.DEPLOY
   },
 
-  isCanceled(deployment) {
-    const action = _.get(deployment, 'deployment.action')
-    return action != null && action.toUpperCase() === this.ACTION_CANCEL
+  isCanceled(process) {
+    const action = _.get(process, 'deployment.action')
+    return action != null && action.toUpperCase() === this.ACTIONS.CANCEL
   },
 
-  //TODO: display e.g. warnings, use it in Visualization (left panel)
-  getStatusClass(processState, shouldRun, statusesLoaded) {
-    const isRunning = _.get(processState, 'isRunning')
-    if (isRunning) {
-      return "status-running"
-    } else if (shouldRun) {
-      return statusesLoaded ? "status-notrunning" : "status-unknown"
+  getStateTooltip(state) {
+    const status = this.getStateStatus(state)
+    return _.get(this.TOOLTIPS, status, this.TOOLTIPS[this.STATUSES.UNKNOWN])
+  },
+
+  getStateIcon(state) {
+    const status = this.getStateStatus(state)
+    return _.get(this.STATUS_ICONS, status, InlinedSvgs.iconUnknown)
+  },
+
+  getStateStatus(state) {
+    if (_.isUndefined(state)) {
+      return this.STATUSES.UNKNOWN
+    }
+
+    if (_.isNull(state)) {
+      return this.STATUSES.CANCELED
     }
-    return null
+
+    return state.status
   },
 
-  getStatusMessage(processState, shouldRun, loaded) {
-    const isRunning = _.get(processState, 'isRunning')
-    if (isRunning) {
-      return "Running"
-    } else if (shouldRun) {
-      const message = (processState || {}).errorMessage || "Not found in engine"
-      return loaded ? `Not running: ${message}` : "Unknown state"
+  getProcessTooltip(process, isStateLoaded) {
+    const pendingTitle = (title) => title + (isStateLoaded === false ? " Loading actually state of process..." : "")
```

Comment:
Loading current state of the process

https://english.stackexchange.com/questions/29654/using-actual-to-signify-current#29655